pkgs/stdenv/generic/check-meta.nix

```diff
@@ -299,7 +300,7 @@ let
 
   checkMetaAttr = k: v:
     if metaTypes?${k} then
-      if metaTypes.${k}.check v then null else "key '${k}' has a value ${toString v} of an invalid type ${builtins.typeOf v}; expected ${metaTypes.${k}.description}"
+      if metaTypes.${k}.check v then null else "key '${k}' has a value of invalid type ${builtins.typeOf v}; expected ${metaTypes.${k}.description}"
```

Comment:
To be honest I don't like a lot about that entire line. The usage of `${builtins.typeOf v}` leads to pretty bad error messages IMO ("Invalid type List expected listof str"). That's one of the reasons I'd like to avoid doing manual type checks like this and use the module system instead

Reply:
https://github.com/NixOS/nixpkgs/pull/180007

but @roberth didn’t like it

Reply:
@Profpatsch My only dislike is that it adds too many parentheses. Those can be reduced a lot by implementing a few rules. I'll be happy to review it again if you or anyone improves it.

Reply:
Another reason against printing the value: It may blow up in the users face if they are accidentally putting some garbage in here, obscuring the actual error message.
https://github.com/NixOS/nixpkgs/pull/168295#issuecomment-1095518417

------

I like #180007, but it won't help against our usage of `${builtins.typeOf v}`